Mathlib/RingTheory/MvPowerSeries/Evaluation.lean

```diff
@@ -0,0 +1,437 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos-Fernández
+-/
+
+import Mathlib.Algebra.MvPolynomial.CommRing
+import Mathlib.RingTheory.Ideal.BigOperators
+import Mathlib.RingTheory.MvPowerSeries.PiTopology
+import Mathlib.RingTheory.MvPowerSeries.Trunc
+import Mathlib.Topology.Algebra.Algebra
+import Mathlib.Topology.Algebra.TopologicallyNilpotent
+import Mathlib.Topology.Algebra.LinearTopology
+import Mathlib.Topology.Algebra.UniformRing
+
+/-! # Evaluation of multivariate power series
+
+Let `σ`, `R` `S` be types, with `CommRing R`, `CommRing S`.
+One assumes that `TopologicalRing R` and `UniformAddGroup R`,
+and that `S` is a complete and separated topological `R`-algebra,
+with `LinearTopology R`, which means there is a basis of neighborhoods of 0
+consisting of ideals.
+
+* `MvPowerSeries.eval₂` : Given `φ : R →+* S` and `a : σ → S`,
+this file defines an evaluation of `f : MvPowerSeries σ R`,
+that extends the evaluation of polynomials at `a`, by density,
+provided `φ` and `a` satisfy certain conditions of which
+the following lemmas assert the necessity
+
+* `Continuous.on_scalars` : The map `φ` is continuous
+* `Continuous.tendsto_apply_pow_zero_of_constantCoeff_zero` :
+  for all `s : σ`, `(a s) ^ n` tends to 0 when `n` tends to infinity
+* `Continuous.tendsto_apply_variables_zero_of_cofinite`:
+  when `a s` tends to  zero for the filter of cofinite subsets of `σ`.
+
+* `MvPowerSeries.eval₂_domain` : the `Prop`-valued structure
+that is required to evaluate power series
+
+* `MvPowerSeries.uniformContinuous_eval₂` : uniform continuity of the evaluation
+
+* `MvPowerSeries.continuous_eval₂` : continuity of the evaluation
+
+* `MvPowerSeries.aeval` : the evaluation map as an algebra map
+
+-/
+
+namespace MvPowerSeries
+
+open Topology
+
+open Filter MvPolynomial RingHom Set TopologicalSpace UniformSpace
+
+/- ## Necessary conditions -/
+
+section
+
+variable {σ : Type*}
+variable {R : Type*} [CommRing R] [TopologicalSpace R]
+variable {S : Type*} [CommRing S] [TopologicalSpace S]
+variable {φ : R →+* S}
+
+-- We endow MvPowerSeries σ R with the Pi topology
+open WithPiTopology
+
+/-- Families at which power series can be evaluated -/
+structure EvalDomain (a : σ → S) : Prop where
```

Comment:
I'm a bit annoyed by the fact that `EvalDomain` heavily suggests a set rather than a predicate. Do you see an alternative name which would sound more like a predicate? If not that's fine.

Reply:
`IsEvaluable`, `HasEval`?

Reply:
The problem of `IsEvaluable` is that I would say a power series is evaluable at a point, not that a point is evaluable. I think I like `HasEval` better.

Reply:
Go for `HasEval`